tensorflow_model_optimization/python/examples/clustering/keras/mnist/mnist_cnn.py

```diff
@@ -41,145 +41,137 @@
-        x_train,
-        y_train,
-        batch_size=batch_size,
-        epochs=epochs,
-        verbose=1,
-        validation_data=(x_test, y_test))
-    score = model.evaluate(x_test, y_test, verbose=0)
-    print('Test loss:', score[0])
-    print('Test accuracy:', score[1])
-
-    print('Clustering model')
-
-    clustering_params = {
-      'number_of_clusters': 8,
-      'cluster_centroids_init': cluster_config.CentroidInitialization.DENSITY_BASED
-    }
-
-    # Cluster model
-    clustered_model = cluster.cluster_weights(model, **clustering_params)
-
-    # Use smaller learning rate for fine-tuning
-    # clustered model
-    opt = tf.keras.optimizers.Adam(learning_rate=1e-5)
-
-    clustered_model.compile(
-    loss=tf.keras.losses.categorical_crossentropy,
-    optimizer=opt,
-    metrics=['accuracy'])
+  return (train_images, train_labels), (test_images, test_labels)
 
-    # Fine-tune model
-    clustered_model.fit(
-        x_train,
-        y_train,
-        batch_size=batch_size,
-        epochs=epochs_fine_tuning,
-        verbose=1,
-        validation_data=(x_test, y_test))
 
-    score = clustered_model.evaluate(x_test, y_test, verbose=0)
-    print('Clustered Model Test loss:', score[0])
-    print('Clustered Model Test accuracy:', score[1])
+def build_sequential_model():
+  "Define the model architecture."
 
-    #Ensure accuracy persists after stripping the model
-    stripped_model = cluster.strip_clustering(clustered_model)
+  return keras.Sequential([
+    keras.layers.InputLayer(input_shape=(28, 28)),
+    keras.layers.Reshape(target_shape=(28, 28, 1)),
+    keras.layers.Conv2D(filters=12, kernel_size=(3, 3), activation='relu'),
+    keras.layers.MaxPooling2D(pool_size=(2, 2)),
+    keras.layers.Flatten(),
+    keras.layers.Dense(10)
+  ])
 
-    stripped_model.compile(
-    loss=tf.keras.losses.categorical_crossentropy,
+
+def train_model(model, x_train, y_train, x_test, y_test):
+  model.compile(
+      loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
+      optimizer='adam',
+      metrics=['accuracy'])
+
+  # Print the model summary.
+  model.summary()
+
+  # Model needs to be clustered after initial training
+  # and having achieved good accuracy
+  model.fit(
+      x_train,
+      y_train,
+      batch_size=batch_size,
+      epochs=epochs,
+      verbose=1,
+      validation_split=0.1)
+
+  score = model.evaluate(x_test, y_test, verbose=0)
+  print('Test loss:', score[0])
+  print('Test accuracy:', score[1])
+  
+  return model
+
+
+def cluster_model(model, x_train, y_train, x_test, y_test):
+  print('Clustering model')
+
+  clustering_params = {
+    'number_of_clusters': 8,
+    'cluster_centroids_init': cluster_config.CentroidInitialization.DENSITY_BASED
+  }
+
+  # Cluster model
+  clustered_model = cluster.cluster_weights(model, **clustering_params)
+
+  # Use smaller learning rate for fine-tuning
+  # clustered model
+  opt = tf.keras.optimizers.Adam(learning_rate=1e-5)
```

Comment:
i've got better accuracy with learning rate 1e-3
 0.9724 (1e-5) vs. 0.9786 (1e-3) 
could be noise, but it trains faster with 1e-3